astropy/coordinates/transformations.py

```diff
@@ -741,8 +744,104 @@ def __call__(self, fromcoord, toframe):
                 'should have been of type {1}'.format(res, self.tosys))
         return res
 
+class BaseAffineTransform(CoordinateTransform):
+    """
+    Base class for common functionality between the ``AffineTransform``-type
+    subclasses.
+    """
+
+    def _apply_transform(self, fromcoord, matrix, vectors):
+        from .representation import (CartesianRepresentation,
+                                     UnitSphericalRepresentation,
+                                     CartesianDifferential)
+
+        rep = fromcoord.represent_as(CartesianRepresentation)
+        newrep = rep.without_differentials()
+
+        # Only do transform is matrix is specified. This is for speed in
+        # transformations that only specify an offset (e.g., LSR)
+        if matrix is not None:
+            newrep = newrep.transform(matrix)
+
+        if vectors is not None and vectors[0] is not None:
+            pos_offset = CartesianRepresentation(vectors[0])
+            newrep = newrep + pos_offset
+
+        if rep.differentials:
+            # assume the 0th item is a velocity
+            # TODO: we only support velocities right now - elsewhere, we need to
+            # validate / make sure only 1 differential is attached
+            veldiff = rep.differentials[0]
+
+            base = fromcoord.data.represent_as(veldiff.base_representation)
+
+            # Only do transform is matrix is specified.
+            newdiff = veldiff.represent_as(CartesianRepresentation, base=base)
```

Comment:
See https://github.com/astropy/astropy/pull/6169#issuecomment-308885139: it would seem handy if for representations that have differentials, at least `to_cartesian` would return a `CartesianRepresentation` that also had the differentials attached, already transformed to `CartesianDifferential`.